core/cache/lxdprofilewatcher.go

```diff
@@ -33,7 +33,7 @@ type MachineAppModeler interface {
 
 type appInfo struct {
 	charmURL     string
-	charmProfile *lxdprofile.Profile
+	charmProfile lxdprofile.Profile
```

Comment:
It is safer to keep this as a value rather than a reference, because it comes from cached `details`.
We can check `Empty()` rather than nil.